src/darwin/Framework/CHIP/ServerEndpoint/MTRIMDispatch.mm

```diff
@@ -83,22 +84,24 @@ void emberAfClusterInitCallback(EndpointId endpoint, ClusterId clusterId)
 
     void DispatchSingleClusterCommand(const ConcreteCommandPath & aPath, TLV::TLVReader & aReader, CommandHandler * aCommandObj)
     {
-        using Protocols::InteractionModel::Status;
+        // TODO: Consider having MTRServerCluster register a
+        // CommandHandlerInterface/ServerClusterInterface for command dispatch.
+        // But OTA would need some special-casing in any case, to call into the
+        // existing cluster implementation.
 
-        if (aPath.mClusterId != OtaSoftwareUpdateProvider::Id) {
-            aCommandObj->AddStatus(aPath, Status::UnsupportedCluster);
-            return;
-        }
+        using Protocols::InteractionModel::Status;
+        using namespace OtaSoftwareUpdateProvider::Commands;
 
-        if (aPath.mEndpointId != kOtaProviderEndpointId) {
+        // This command passed ServerClusterCommandExists so we know it's one of our
```

Comment:
There is no ServerClusterCommandExists() anymore.  It passed `InteractionModelEngine::ValidateCommandCanBeDispatched`.